optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendable/BendableScoreHibernateTypeTest.java

```diff
@@ -35,6 +35,22 @@ public void persistAndMerge() {
                 BendableScore.ofUninitialized(-7, new int[]{10000, 2000, 300}, new int[]{40, 5}));
     }
 
+    @Test
+    public void persistUninitializedScoreThenFind() {
```

Comment:
We'll want every ScoreHibernateTypeTest to test for this gap, not just the BendableScore one.

But to make it easier, can we not just do the find as part of the persistAndMerge method (and maybe rename it to persistMergeAndFind() ?)

Reply:
Actually the exception is thrown at [this line](https://github.com/kiegroup/optaplanner/blob/e773f20110e589e6fabcfec0e2a893db7fe0f1cf/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateTypeTest.java#L64) which is also called by the [existing persistAndMerge()](https://github.com/kiegroup/optaplanner/blob/e773f20110e589e6fabcfec0e2a893db7fe0f1cf/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendable/BendableScoreHibernateTypeTest.java#L32) test. When it's called with `new TestJpaEntity(null)` no exception is thrown although the same method `em.persist(jpaEntity)` is called again several times with non null scores [here](https://github.com/kiegroup/optaplanner/blob/e773f20110e589e6fabcfec0e2a893db7fe0f1cf/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateTypeTest.java#L93) with no exceptions thrown either. When I modified the existing test and replaced the `null` score with BendableScore.zero(...) I got the exception.  I need to investigate more why this is happening to fix it properly.